client/src/components/CostChart.tsx

```diff
@@ -0,0 +1,496 @@
+import { useEffect, useMemo, useRef, useState } from "react";
+
+import type { ResourceCost, WorkspaceResource } from "../types/api";
+
+interface CostChartProps {
+  costData: ResourceCost[];
+  selectedResources: WorkspaceResource[];
+  loadingResources?: string[];
+}
+
+type ViewType = "table" | "chart";
+
+type TimeSeriesEntry = {
+  date: string;
+  [resourceName: string]: string | number;
+};
+
+function ViewSelector({
```

Comment:
can you move it out as a separate component pls?

Reply:
Yes, moved all the components to more appropriate places